### DISCUSSION TITLE
socket.leaveAll in 4.x?

### DISCUSSION THREAD

**Discussion by Rc85** [socketio/socket.io#3932] [Q&A]
What happen to this function? Must all channels be iterated over to leave now?

**Comment by darrachequesne** [comment]
Hi! The `leaveAll` method is [private](https://github.com/socketio/socket.io/blob/1633150b2bcdebd3c39dc37f7329e84e9951a15e/lib/socket.ts#L320-L327) (this was already the case in v2).

You shouldn't have to call it directly though, as it is automatically called [upon disconnection](https://github.com/socketio/socket.io/blob/1633150b2bcdebd3c39dc37f7329e84e9951a15e/lib/socket.ts#L482). Could you please explain your use case?

**Reply by Rc85** [reply]
I just wanted to make sure the user leaves all the rooms before joining another and `leaveAll` was just easier. I was using JS before so it didn't warn me about method being private, TS does now.

**Reply by darrachequesne** [reply]
I don't think we should make it `public`, since it will lead to some weird behaviors. You should be able to leave all rooms with:

```js
socket.rooms.forEach((room) => {
  if (room !== socket.id) {
    socket.leave(room);
  }
});
```
